dbms/src/Common/DNSResolver.cpp

```diff
@@ -128,14 +150,52 @@ String DNSResolver::getHostName()
     if (impl->disable_cache)
         return Poco::Net::DNS::hostName();
 
-    std::unique_lock lock(impl->mutex);
+    std::lock_guard lock(impl->drop_mutex);
 
     if (!impl->host_name.has_value())
         impl->host_name.emplace(Poco::Net::DNS::hostName());
 
+    addToNewHosts(*impl->host_name);
     return *impl->host_name;
 }
 
+bool DNSResolver::updateCache()
+{
+    {
+        std::lock_guard lock(impl->drop_mutex);
+        for (auto & host : impl->new_hosts)
+            impl->known_hosts.insert(std::move(host));
+        impl->new_hosts.clear();
+    }
+
+    std::lock_guard lock(impl->update_mutex);
+
+    bool updated = false;
+    for (const auto & host : impl->known_hosts)
+    {
+        try
+        {
+            updated |= updateHost(host);
+        }
+        catch (const Poco::Net::NetException &) {}
```

Comment:
Why not to log the error?

Reply:
I think, there will be too many such errors in log.

Reply:
I still think that failure in DNS resolution should be logged. Let's look.
Also please add ProfileEvent.